### PR TITLE
feat(attn): running-max softmax for FlyDSL BF16 dualwave forward

### PR DESCRIPTION
**Pull request by kylasa**
Causal dualwave no longer defaults to a fixed zero reference max, which overflows fp32 exp2 in Llama-3.1-8B training. Forward now tracks a real per-tile (m, l, O) running max. Unset env keeps this path; the calibrated constant ref-max remains an inert opt-in.

# Description

Please include a brief summary of the changes, relevant motivation and context.

Fixes # (issue)

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refactoring

## Changes

Please list the changes introduced in this PR:

- Change A
- Change B

# Checklist:

- [ ] The functionality is complete
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: primus_turbo/pytorch/kernels/attention/attention_flydsl_impl.py [hunk 1]
```diff
@@ -30,6 +31,38 @@
...
 
+# Opt-in per-call isfinite(LSE) tripwire. Off by default (one extra reduction).
+# Causal dualwave now defaults to a real running-max softmax; the old fixed-zero
+# reference overflows fp32 exp2 once a row's logit crosses ~88.7 nats.
+ENV_FLYDSL_LSE_GUARD = "PRIMUS_TURBO_FLYDSL_LSE_GUARD"
+
+
+@functools.lru_cache(maxsize=1)
+def _lse_guard_enabled() -> bool:
+    return os.environ.get(ENV_FLYDSL_LSE_GUARD, "0") == "1"
```
Comment (RuibinCheung) [inline primus_turbo/pytorch/kernels/attention/attention_flydsl_impl.py:42]: Remove this environmental variable.

**Review by RuibinCheung — CHANGES_REQUESTED** [review]
